### PR TITLE
feat: Add DELETE device resource test

### PR DESCRIPTION
**Pull request by jumpingliu**
feat #621

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Negative-Deviceresource.robot [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+*** Variables ***
+${SUITE}          Core Metadata Device Profile DELETE Device Resource Negative Test Cases
+${LOG_FILE_PATH}  ${WORK_DIR}/TAF/testArtifacts/logs/core-metadata-deviceprofile-delete-deviceresource-negative.log
+
+*** Test Cases ***
+ErrProfileResourceDELETE001 - Delete deviceResource by non-existent deviceResource name
+    # non-existent deviceResource name
+    Given Create A Device Profile Sample With non-existent deviceResource name
+    When Delete deviceResource By Name
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Negative-Deviceresource.robot:18]: Should create a normal device profile and delete device resource by non-existent deviceResource name

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Negative-Deviceresource.robot [hunk 2]
```diff
@@ -0,0 +1,39 @@
+*** Settings ***
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Negative-Deviceresource.robot:1]: Should add [Teardown] to each testcase to remove device profile created

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Negative-Deviceresource.robot [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    Then Should Return Status Code "404"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ErrProfileResourceDELETE002 - Delete deviceResource which profile used by device
+    Given Create A Device Profile Sample With Associated Test-Device
+    When Delete deviceResource By Name
+    Then Should Return Status Code "409"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Negative-Deviceresource.robot:28]: Should delete device profile and the assicated device created in this testcase

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Positive-Deviceresource.robot [hunk 4]
```diff
@@ -0,0 +1,31 @@
+*** Settings ***
+Resource     TAF/testCaseModules/keywords/common/commonKeywords.robot
+Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
+Suite Setup  Run Keywords  Setup Suite
+...                        AND  Run Keyword if  $SECURITY_SERVICE_NEEDED == 'true'  Get Token
+Suite Teardown  Run Teardown Keywords
+Force Tags      v2-api
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Positive-Deviceresource.robot:7]: Use `skipped` instead.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Positive-Deviceresource.robot [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+*** Test Cases ***
+ProfileResourceDELETE001 - Delete deviceResource on one device profile with deviceResource
+    Given Create A Device Profile with deviceResource
+    When Delete deviceResource By Name
+    Then Should Return Status Code "200"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And Device Resource Should Be Deleted
+
+ProfileResourceDELETE001 - Delete deviceResource on device profile basicinfo only
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-Positive-Deviceresource.robot:22]: This scenario is same as the one above, so this testcase can be removed.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

**Comment by jumpingliu** [comment]
Drop the commit so reopen the PR.

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**